File: src/Controller.cpp
#include <Controller.hpp>
#include <SDL2/SDL.h>

Buttons GetButtons()
{
    uint8_t const* keyStates = SDL_GetKeyboardState(nullptr);
    Buttons buttons;

    buttons.down = keyStates[SDL_SCANCODE_S];
    buttons.up = keyStates[SDL_SCANCODE_W];
    buttons.left = keyStates[SDL_SCANCODE_A];
    buttons.right = keyStates[SDL_SCANCODE_D];
    buttons.start = keyStates[SDL_SCANCODE_RETURN];
    buttons.select = keyStates[SDL_SCANCODE_RSHIFT];
    buttons.b = keyStates[SDL_SCANCODE_K];
    buttons.a = keyStates[SDL_SCANCODE_L];

    return buttons;
}

File: GUI/GUI/controller.py
from dataclasses import dataclass
from sdl2 import *

@dataclass
class JoyPad:
    """Current joypad state."""
    down: bool
    up : bool
    left: bool
    right: bool
    start: bool
    select: bool
    b: bool
    a: bool

def get_joypad() -> JoyPad:
    key_states = SDL_GetKeyboardState(None)
    return JoyPad(key_states[SDL_SCANCODE_S],
                  key_states[SDL_SCANCODE_W],
                  key_states[SDL_SCANCODE_A],
                  key_states[SDL_SCANCODE_D],
                  key_states[SDL_SCANCODE_RETURN],
                  key_states[SDL_SCANCODE_RSHIFT],
                  key_states[SDL_SCANCODE_K],
                  key_states[SDL_SCANCODE_L])
